recommendations for function comments

According to the rules of Go, exported functions should have comments, but they are not necessary for Exercism exercises, and most submissions do not include them.

In [hello-world](https://github.com/exercism/xgo/blob/master/exercises/hello-world/hello_world.go) we say:

> HelloWorld needs a comment documenting it as package does.

and

> Package greeting should have a package comment that summarizes what it's about.

Should we change this to a recommendation for what is good Go practice, but that for these exercises they're not necessary? 

Similar to what is currently in `leap`:

>  It's good style to write a comment here documenting IsLeapYear.

We're not going to enforce comments and it's up to each individual how they want to approach these exercises; as a way to practice writing code, or as a way to build a habit of best practices.

I wondered if it would be good to automate something like that, so that it doesn't become something that gets easily out of sync.
I used the following script in the root of the repo:

```sh
for d in exercises/*/ ; do                                                                                                                 
     golint ./$d | grep -v comment                                                                                                          
done
```

This ignores any golint complaints that are about comments, which is nearly all of them, the rest is:

```sh                                                                                                                         [127]
exercises/clock/clock.go:14:12: expected type, found newline (and 1 more errors)
exercises/gigasecond/gigasecond.go:5:21: illegal character U+003F '?' (and 1 more errors)
exercises/palindrome-products/example.go:50:20: error strings should not be capitalized or end with punctuation or a newline
exercises/roman-numerals/example.go:19:25: error strings should not be capitalized or end with punctuation or a newline
exercises/triangle/triangle.go:9:10: expected type, found newline (and 1 more errors)
```

It's perhaps worth moving this into a new issue?

--- a/exercises/hello-world/hello_world.go
+++ b/exercises/hello-world/hello_world.go
@@ -3,6 +3,12 @@
 
 // Package greeting should have a package comment that summarizes what it's about.
 // https://golang.org/doc/effective_go.html#commentary
+// While documentation is taken very seriously in Go, and tools like golint will
+// complain if you don't include it, these comments are not required to make
+// the test suite pass. It is up to you whether to include them in your submitted
+// exercises.
+// You can read more about the importance of documentation in Go here:
+// https://blog.golang.org/godoc-documenting-go-code
 package greeting
 
 // testVersion identifies the version of the test program that you are
@@ -22,7 +28,10 @@ package greeting
 // information.
 const testVersion = 4
 
-// HelloWorld needs a comment documenting it as package does.
+// HelloWorld should have a comment documenting it, beginning with the name of
+// the function. It is recommended good practice, but, as we mention above, it's
+// not necessary for making your solutions pass.
+// https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences
 func HelloWorld() string {
 	// Write some code here to pass the test suite.
 

